src/tblite_types.F

```diff
@@ -0,0 +1,125 @@
+!                                                                                                  !
+!   SPDX-License-Identifier: GPL-2.0-or-later                                                      !
+!--------------------------------------------------------------------------------------------------!
+
+! **************************************************************************************************
+!> \brief types for tblite
+!> \author JVP
+!> \history creation 09.2024
+! **************************************************************************************************
+
+MODULE tblite_types
+
+#if defined(__TBLITE)
+   USE mctc_io, ONLY: structure_type, new
+   USE tblite_xtb_calculator, ONLY: xtb_calculator
+   USE tblite_container, ONLY: container_cache
+   USE tblite_wavefunction_type, ONLY: wavefunction_type
+   USE tblite_scf_potential, ONLY: potential_type
+#endif
+
+   USE cp_dbcsr_api, ONLY: dbcsr_p_type
+   USE cp_dbcsr_operations, ONLY: dbcsr_deallocate_matrix_set
+   USE kinds, ONLY: dp
+
+#include "./base/base_uses.f90"
+   IMPLICIT NONE
+
+   PRIVATE
+
+   CHARACTER(len=*), PARAMETER, PRIVATE :: moduleN = 'tblite_types'
+
+   PUBLIC :: tblite_type, deallocate_tblite_type, allocate_tblite_type
+
+   TYPE tblite_type
+
+      INTEGER, ALLOCATABLE                               :: el_num(:)
+
+      REAL(KIND=dp), DIMENSION(3, 3)                     :: sigma
+      REAL(KIND=dp), ALLOCATABLE                         :: e_hal(:)
+      REAL(KIND=dp), ALLOCATABLE                         :: e_rep(:)
+      REAL(KIND=dp), ALLOCATABLE                         :: e_disp(:)
+      REAL(KIND=dp), ALLOCATABLE                         :: e_scd(:)
+      REAL(KIND=dp), ALLOCATABLE                         :: e_es(:)
+      REAL(KIND=dp), ALLOCATABLE                         :: selfenergy(:)
+      REAL(KIND=dp), ALLOCATABLE                         :: cn(:)
+
+      REAL(KIND=dp), ALLOCATABLE                         :: grad(:, :)
+      REAL(KIND=dp), ALLOCATABLE                         :: dsedcn(:)
+      REAL(KIND=dp), ALLOCATABLE                         :: dcndr(:, :, :)
+      REAL(KIND=dp), ALLOCATABLE                         :: dcndL(:, :, :)
+
+      TYPE(dbcsr_p_type), DIMENSION(:), POINTER          :: dipbra => Null(), &
+                                                            dipket => Null()
+      TYPE(dbcsr_p_type), DIMENSION(:), POINTER          :: quadbra => Null(), &
+                                                            quadket => Null()
+
+#if defined(__TBLITE)
+      TYPE(structure_type)                               :: mol
+      TYPE(xtb_calculator)                               :: calc
+      TYPE(container_cache)                              :: cache
+      TYPE(container_cache)                              :: dcache
+      TYPE(wavefunction_type)                            :: wfn
+      TYPE(potential_type)                               :: pot
```

Comment:
Can you try to add a default initializer here? Use something like `TYPE(my_type) :: var = my_type()`. At least for some types, there are defaults defined.